Intel_SighPP_CMake/Intel_SighPP_CMake/src/Intel_Sight_++.cpp

```diff
@@ -172,14 +212,27 @@ int main(int argc, char** argv)
 	// TODO Add command line parameter for files to use?
 
 	auto profile = config.get_stream(RS2_STREAM_COLOR).as<rs2::video_stream_profile>();
-	CaffeModelImpl caffe_own("./models/no_bn.prototxt", "./models/no_bn.caffemodel", "./models/no_bn-classnames.txt");
-	CaffeModelImpl caffe_pre("./models/MobileNetSSD_deploy.prototxt", "./models/MobileNetSSD_deploy.caffemodel", "./models/MobileNetSSD_deploy-classnames.txt");
+
+	for (auto i = caffe_models.begin(); i != caffe_models.end(); ++i)
```

Comment:
Instead of for loops, you could also move the ML controller creation to above the if (argc > 1) statement, and add directly to it.

Reply:
For some reasons you cannot put `CaffeModelImpl caffe_model(prototxt_path_, caffemodel_path_, txt_path_)` and ml_controller.add_model(caffe_model) together inside a loop. For example you cant have 
```
for (auto i = caffe_models.begin(); i != caffe_models.end(); ++i)
	{
                CaffeModelImpl caffe_model(prototxt_path_, caffemodel_path_, txt_path_)
		ml_controller.add_model(caffe_model);
	}
```
and it will give segmentation fault 11 on my laptop if you do so. Therefore you have to separate these two lines instead of putting them together inside a for-loop. I know it's weird and I did what you proposed in the very beginning but it gave error. Songping also checked it and he had the same result. So that's why I used a for-loop here.

Reply:
Hmm, weird. Okay then 😄